manticore/native/memory.py

```diff
@@ -814,6 +814,69 @@ def __str__(self):
             ]
         )
 
+    def proc_self_mappings(self):
+        """
+        Returns a sorted list of all the mappings for this memory for /proc/self/maps.
+        Device, inode, and private/shared permissions are unsupported.
+        Stack is the only memory section supported in the memory map (heap, vdso, etc.)
+        are unsupported.
+        Pathname is substituted by filename
+
+        :return: a list of mappings.
+        :rtype: list
+        """
+        result = []
+        # TODO: Device, inode, and private/shared permissions are unsupported
+        # TODO: Add complete paths
+        device = "00:00"
+        inode = 0
+        private_shared_perms = "-"
+        for m in self.maps:
+            if isinstance(m, AnonMap):
+                if m.name is not None:
+                    result.append(
+                        (
+                            m.start,
+                            m.end,
+                            m.perms + private_shared_perms,
+                            0,
+                            device,
+                            inode,
+                            "[" + m.name + "]",
+                        )
+                    )
```

Comment:
Could we make this into something more well-defined?

Perhaps a `@dataclass` class that has useful names.

Something like
```python
@dataclass
class ProcSelfMapInfo
    start_addr: int
    end_addr: int
    map_perms: ....

[...]

result.append(ProcSelfMapInfo(m.start, m.end, m.perms + private_shared_perms, ...))
```
More info about dataclasses https://pypi.org/project/dataclasses/ (PyPI link since we support Python3.6)

Reply:
Will take a look at this :)